src/App.tsx

```diff
@@ -15,33 +16,123 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+// eslint-disable-next-line max-len
+function getReorderedGoods(goods: string[], sortType: SortType, isReversed: boolean) {
+  const visibleGoods = [...goods].sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+let activeButtons: string[] = [];
+
+export const App: React.FC = () => {
+  const [showGoods, setShowGoods] = useState(false);
+  const [sortType, setSortType] = useState(0);
+  const [reverse, setReverse] = useState(false);
+  const preparedGoods
+    = getReorderedGoods(goodsFromServer, sortType, reverse);
+
+  const setActiveButton = (addElement: string, removeElements = false) => {
+    if (removeElements) {
+      activeButtons = [];
+    }
+
+    if (!activeButtons.includes(addElement)) {
+      activeButtons.push(addElement);
+    } else if (addElement === 'reverse' && activeButtons.includes(addElement)) {
+      const index = activeButtons.indexOf(addElement);
+
+      activeButtons.splice(index, 1);
+    }
+  };
+
+  return (
+    <div className="App">
+      {(!showGoods
+        && (
+          <button
+            type="button"
+            className="button"
+            onClick={() => (
+              setShowGoods(true)
+            )}
+          >
+            Start
+          </button>
+        )) || (
```

Comment:
Strange condition fixes it pls. If u want to use ternary operator use this template pls

```
{condition
? (
		jsx
)
: (
	jsx
)

```